### ISSUE TITLE
Gradle 3.0 alpha 5, Android Studio 3.0 preview

### ISSUE THREAD

**Issue by DaanGeurts** [bugsnag/bugsnag-android-gradle-plugin#40]
Thank you for all your work so far. It helped us a lot with our products. 
I'm investigating the new Android Studio 3.0 preview with Gradle 3.0.0-alpha5 but I have an error building our app.
I'm using version 2.3.2 of this plugin.

> Error:Could not get unknown property 'manifestOutputFile' for task ':app:processDebugManifest' of type com.android.build.gradle.tasks.MergeManifests.

When I disable this plugin, the gradle build does work again.
Thanks in advance!

**Comment by kattrali** [comment]
Thanks for the report, @DaanGeurts. This is probably a similar issue to #39 -- investigating but we might need gradle 3/new Android Studio to stabilize a bit first.

**Comment by fractalwrench** [comment]
It looks like the Variant API is broken in the latest release of the Android Gradle plugin. This means our plugin is able to get the outputFile for the AndroidManifest, and therefore can't unable to add a Build UUID for the variant being built.

Google have stated that they will add new APIs to provide similar functionality, but unfortunately there doesn't seem to be any timeline on that yet.

https://developer.android.com/studio/preview/features/new-android-plugin-migration.html#variant_api

**Comment by jbrick** [comment]
@fractalwrench I was talking to someone on the Android Tools team about this.

"There is now a single manifest task per variant. You can get access to it and do the same as you do now. However its output is a folder with all the different manifests in it (in case you have multi-apk/splits)"

I think you probably just need to modify this to expect a folder as output instead of a single manifest file.

**Comment by artem-zinnatullin** [comment]
Any update/workaround on that? 

Our migration to Android Gradle Plugin 3.x is blocked by Bugsnag Gradle Plugin :(

**Comment by fractalwrench** [comment]
A temporary workaround is available on [this feature branch](https://github.com/bugsnag/bugsnag-android-gradle-plugin/tree/android-gradle-plugin-3-alpha), which should allow gradle to sync successfully.

*However*, Proguard map files will not be uploaded when generating a clean build of each variant. This is because the variant task no longer knows the manifest file location up front.

We'll continue to investigate a more permanent solution to this problem. In the meantime if this is blocking you, it should be possible to install the workaround version of the plugin in a [local repository](https://github.com/bugsnag/bugsnag-android-gradle-plugin/blob/master/CONTRIBUTING.md).

**Comment by artem-zinnatullin** [comment]
Great news! 

So you suggest to run `uploadBugsnag` tasks again after clean build to actually upload mapping?

**Comment by kattrali** [comment]
Posting the upstream issue for visibility: https://issuetracker.google.com/issues/64747519

**Comment by Benlenem** [comment]
Hi everyone,

Is the issue fixed ? I'm migrating a project to latest gradle (3.1.2) and bugsnag  plugin version (3.8.0).
I still have the issue described above

```
Caused by: java.lang.RuntimeException: Manifest Tasks does not support the manifestOutputFile property any more, please use the manifestOutputDirectory instead.
For more information, please check https://developer.android.com/studio/build/gradle-plugin-3-0-0-migration.html
```

**Comment by fractalwrench** [comment]
@Benlenem would you be able to post some details of your project configuration in a separate issue? Versions 3.0.0 and upwards of the bugsnag plugin should be compatible with AGP 3+.

**Comment by Benlenem** [comment]
I was still using a 2.+ version of the plugin. It works fine with 3+. Sorry for the disturbance
